src/bootstrap/compile.rs

```diff
@@ -1106,7 +1106,7 @@ impl Step for Sysroot {
         let compiler = self.compiler;
         let host_dir = builder.out.join(&compiler.host.triple);
         let sysroot = if compiler.stage == 0 {
-            host_dir.join("stage0-sysroot")
+            host_dir.join("bootstrap-sysroot")
```

Comment:
Ok, I've found the issue. `stage0-sysroot` is *not* being replaced with `bootstrap-sysroot`, it's the sysroot where the *generated* libstd artifacts are placed, it doesn't have the beta artifacts. The rust code is no longer using bootstrap-sysroot at all except for `initial_rustc`. So actually it's become consistent with the final branch below on line 1113 and stays as `stage0-sysroot`, and you can remove this branch of the code altogether.

The error you were seeing was because bootstrap deletes the sysroot before it generates it to avoid caching problems; in this case it was deleting the downloaded sysroot, which meant `initial_rustc` no longer existed.

Reply:
I think to avoid problems like this cropping up again, you should add a new `fn sysroot_dir(config: &Config, stage: u8) -> PathBuf` that we use *anywhere* that currently uses a sysroot, and then we can be sure we're not doing this wrong.

Reply:
ok so tried doing this to get the build working , but seems its not happy with it.

```
 let sysroot = if compiler.stage == 0 {
            host_dir.join("bootstrap-sysroot")
        } else
        /*if builder.download_rustc()*/
        {
            host_dir.join("ci-rustc-sysroot")
        };

        //else {
        //    host_dir.join(format!("stage{}", compiler.stage))
        //};

```

Reply:
No, other way around: remove the branch for `stage == 0` and keep the rest.
```rust
let sysroot = if builder.download_rustc() {
            host_dir.join("ci-rustc-sysroot")
        } else {
            host_dir.join(format!("stage{}", compiler.stage))
        };
```

Reply:
Again, `download_rustc` shouldn't be changing anywhere in this PR.

Reply:
Thank you . That worked .. lets see if the build goes thru and then any code review comments.